qiskit/opflow/gradients/circuit_qfis/lin_comb_full.py

```diff
@@ -31,33 +32,50 @@ class LinCombFull(CircuitQFI):
     r"""Compute the full Quantum Fisher Information (QFI).
 
     Given a pure, parameterized quantum state this class uses the linear combination of unitaries
-    approach, requiring one additional working qubit.
     See also :class:`~qiskit.opflow.QFI`.
     """
 
+    # pylint: disable=signature-differs
+    # pylint: disable=arguments-differ
     def convert(
         self,
         operator: CircuitStateFn,
         params: Union[ParameterExpression, ParameterVector, List[ParameterExpression]],
+        aux_meas_op: OperatorBase = Z,
+        phase_fix: bool = True,
     ) -> ListOp:
         r"""
         Args:
             operator: The operator corresponding to the quantum state :math:`|\psi(\omega)\rangle`
                 for which we compute the QFI.
             params: The parameters :math:`\omega` with respect to which we are computing the QFI.
+            aux_meas_op: The operator that the auxiliary qubit is measured with respect to.
+                For ``aux_meas_op = Z`` we compute 4Re[⟨(dω⟨ψ(ω)|)O(θ)|ψ(ω)〉],
+                for ``aux_meas_op = -Y`` we compute 4Im[⟨(dω⟨ψ(ω)|)O(θ)|ψ(ω)〉], and.
+                for ``aux_meas_op = Z-1jY`` we compute 4⟨(dω⟨ψ(ω)|)O(θ)|ψ(ω),
+            phase_fix: Whether or not to compute the additional phase fix term
+                Re[(dω⟨<ψ(ω)|)|ψ(ω)><ψ(ω)|(dω|ψ(ω))>].
+
 
         Returns:
             A ``ListOp[ListOp]`` where the operator at position ``[k][l]`` corresponds to the matrix
             element :math:`k, l` of the QFI.
 
         Raises:
             TypeError: If ``operator`` is an unsupported type.
+            ValueError: If the provided auxiliary measurement operator is not supported.
         """
         # QFI & phase fix observable
-        qfi_observable = ~StateFn(4 * Z ^ (I ^ operator.num_qubits))
-        phase_fix_observable = StateFn(
-            (Z + 1j * Y) ^ (I ^ operator.num_qubits), is_measurement=True
-        )
+        qfi_observable = StateFn(4 * aux_meas_op ^ (I ^ operator.num_qubits), is_measurement=True)
+        if aux_meas_op not in [Z, -Y, (Z - 1j * Y)]:
+            raise ValueError(
+                "This auxiliary measurement operator is currently not supported please choose "
+                "either Z, -Y, or Z-1jY. "
+            )
+        if phase_fix:
+            phase_fix_observable = SummedOp(
+                [Z ^ (I ^ operator.num_qubits), -1j * Y ^ (I ^ operator.num_qubits)]
+            )
```

Comment:
Do I see correctly that this `if` could be merged with the `if` below starting at line 94?